addons/website/static/src/client_actions/website_preview/website_preview.js

```diff
@@ -338,7 +338,44 @@ export class WebsitePreview extends Component {
         this.title.setParts({ action: currentTitle });
     }
 
-    _onPageLoaded() {
+    _onPageLoaded(ev) {
+        // FIX Chrome-only. If you have the backend in a language A but the
+        // website in English only, you can 1) modify a record's (event,
+        // product...) name in language A (say "New Name").
+        // 2) visit the page `/new-name-11` => the server will redirect you to
+        // the English page `/origin-11`, which is the only one existing.
+        // Chrome caches the redirection.
+        // 3) give the same name in English as in language A, try to visit
+        // => the server now wants to access `/new-name-11`
+        // => Chrome uses the cache to redirect `/new-name-11` to `/origin-11`,
+        // => the server tries to redirect to `/new-name-11` => start again 🔄.
+        // Chrome injects a "Too many redirects" layout in the iframe, which in
+        // turn raises a CORS error when the app tries to update the iframe.
+        // If we detect that behavior, we reload the iframe with a new query
+        // parameter, so that it's not cached for Chrome.
+        try {
+            /* eslint-disable no-unused-expressions */
+            this.iframe.el.contentWindow.location.href;
+        } catch (err) {
+            if (this.iframe.el.src.includes("iframe_reload")) {
+                // Reloading with new param did not solve the redirect cache
+                // issue. The security error could just be about a PDF from
+                // same domain caught as SecurityError on Firefox. Other
+                // browsers seems to recognize that it's actually a PDF from
+                // the same origin and don't raise when accessing the iframe.
+                return;
+            } else {
+                if (err.name === "SecurityError") {
+                    ev.stopImmediatePropagation();
+                    const srcUrl = new URL(this.iframe.el.src);
+                    const pathUrl = new URL(`${srcUrl.origin}${srcUrl.searchParams.get("path")}`);
+                    pathUrl.searchParams.set("iframe_reload", "1");
+                    srcUrl.searchParams.set("path", `${pathUrl.pathname}${pathUrl.search}`);
+                    this.iframe.el.src = srcUrl.toString();
+                    return;
+                }
+            }
+        }
```

Comment:
@rdeodoo To avoid the 3 oneliners, you could just call `ev.stopImmediatePropagation();` like I did.
That said, I tested your fix and what it does is cancelling the pdf loading and going back to the home page. If the link was on the homepage and you click again on the link, then it works. Which made me think of this (very ugly) fix.
Advantage: it "works".
Disadvantage: it flickers.

```diff
        try {
            /* eslint-disable no-unused-expressions */
            this.iframe.el.contentWindow.location.href;
        } catch (err) {
            if (this.iframe.el.src.includes("iframe_reload")) {
+                ev.stopImmediatePropagation();
                // Reloading with new param did not solve the redirect cache
                // issue. The security error could just be about a PDF from
                // same domain caught as SecurityError on Firefox. Other
                // browsers seems to recognize that it's actually a PDF from
                // the same origin and don't raise when accessing the iframe.
                return;
            } else {
                if (err.name === "SecurityError") {
                    ev.stopImmediatePropagation();
                    const srcUrl = new URL(this.iframe.el.src);
                    const pathUrl = new URL(`${srcUrl.origin}${srcUrl.searchParams.get("path")}`);
                    pathUrl.searchParams.set("iframe_reload", "1");
                    srcUrl.searchParams.set("path", `${pathUrl.pathname}${pathUrl.search}`);
+                    if (navigator.userAgent.includes("Firefox")) {
+                        history.back();
+                        setTimeout(() => history.forward(), 50);
+                    }
                    this.iframe.el.src = srcUrl.toString();
                    return;
                }
            }
        }
```

Reply:
That doesn't seems to work on my end, I still got a reload to the homepage (or iframe src to be exact, could be anything, the URL first loaded in the iframe). Second click works, but same as before: when you click on "Website" from the Odoo topbar, it just crashed on cors error again.

I don't think we can make it work, and we can't even load it to the top window because once the iframe is loaded, we can't know its URL (unless we store it before).
Indeed, to know the iframe url (to forward to new tab or top parent window), we have to get it through `contentWindow.location.href` and not `src` (browser spec, iframe src is not updated with navigation) and we can't do that.

We can't neither make it work by listening the ondomloaded or onload in the iframe directly, as the browser takes over the iframe with its own layout/DOM.

I am short on solutions, the only way would seem to be doing some sort of preprocessing before clicking on links which is quite violent (slow down everything / every navigation).

Let's ignore this for now and keep the PR to what it was: solving the redirect cache issue in iframe.